net/src/interface/address.rs

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+use std::fmt::Display;
+use std::net::IpAddr;
+use thiserror::Error;
+
+/// The type of error returned when building an `IfAddr`
+/// if the ip address or mask are not legal.
+#[derive(Debug, Error, PartialEq)]
+pub enum IfAddrError {
+    #[error("Invalid interface mask length '{0}'")]
+    InvalidMask(u8),
+
+    #[error("Invalid interface address '{0}'")]
+    InvalidAddress(IpAddr),
+}
+
+/// An Ipv4 or Ipv6 address and mask configured on an interface
+#[derive(Debug, Clone, Copy, Eq, Hash, PartialEq)]
+pub struct IfAddr {
+    address: IpAddr,
+    mask_len: u8,
+}
+impl IfAddr {
+    /// Tell if an `IpAddr` plus a mask length are suitable to be configured on a network interface.
+    ///
+    /// # Errors
+    ///
+    /// This function returns `IfAddrError` if the provided address is not suitable
+    /// for a network interface or the mask is not legal.
+    pub fn is_valid_ifaddr(address: IpAddr, mask_len: u8) -> Result<(), IfAddrError> {
+        if address.is_multicast() || address.is_unspecified() {
+            Err(IfAddrError::InvalidAddress(address))
+        } else if address.is_ipv4() && mask_len > 32
+            || address.is_ipv6() && mask_len > 128
+            || mask_len == 0
+        {
+            Err(IfAddrError::InvalidMask(mask_len))
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Create an Ipv4 or Ipv6 address and mask length to be configured on an interface.
+    ///
+    /// # Errors
+    ///
+    /// Same as those of `IfAddr::is_valid_ifaddr`
```

Comment:
We can probably use a rustdoc link here, which would catch future name changes for the method?

```suggestion
    /// Same as those of [`is_valid_ifaddr`]
```